relay-server/src/actors/spooler/mod.rs

```diff
@@ -662,6 +666,26 @@ pub struct BufferService {
 impl BufferService {
     /// Set up the database and return the current number of envelopes.
     async fn setup(path: &PathBuf) -> Result<(), BufferError> {
+        if path.is_absolute() {
+            let mut dir = path.clone();
+            dir.pop();
+
+            if !dir.exists() {
+                relay_log::debug!("creating directory for spooling file: {}", dir.display());
+                DirBuilder::new()
+                    .recursive(true)
+                    .create(dir)
+                    .await
+                    .map_err(|e| BufferError::BufferSetupFailed(e.to_string()))?;
+            }
+        } else {
+            // sqlx doesn't support relative paths and Sqlite initialization
+            // fails. Maybe: https://github.com/launchbadge/sqlx/issues/1260
+            return Err(BufferError::BufferSetupFailed(
+                "relative paths for spooling are not supported".to_string(),
+            ));
+        }
```

Comment:
how was this tested? 
I've ran relay locally and the relative paths working just fine. 

Could you, please, post your test or config, so I could run it as well?

Reply:
`~` is a shell thingy and won't be expanded to user's home directory, and will be treated as the literal. 

As I posted before the relative paths working just fine. And we do not need any extra check for those.

Reply:
You're right, removed this piece of code. I clarified the behavior of `~` in the docs PR in the description.